src/Rpc/NodeGuardService.cs

```diff
@@ -250,4 +271,201 @@ ISchedulerFactory schedulerFactory
             throw new RpcException(new Status(StatusCode.Internal, e.Message));
         }
     }
+
+    public override async Task<AddNodeResponse> AddNode(AddNodeRequest request, ServerCallContext context)
+    {
+        var node = new FundsManager.Data.Models.Node
+        {
+            PubKey = request.PubKey,
+            Name = request.Name,
+            Description = request.Description,
+            ChannelAdminMacaroon = request.ChannelAdminMacaroon,
+            Endpoint = request.Endpoint,
+            AutosweepEnabled = request.AutosweepEnabled,
+            ReturningFundsWalletId = request.ReturningFundsWalletId,
+        };
+
+        try
+        {
+            var result = await _nodeRepository.AddAsync(node);
+
+            if (result.Item1)
+            {
+                return new AddNodeResponse();
+            }
+
+            _logger?.LogError("Error adding node, error: {error}", result.Item2);
+            throw new RpcException(new Status(StatusCode.Internal, "Error adding node"));
+        }
+        catch (Exception e)
+        {
+            _logger?.LogError(e, "Error getting adding node through gRPC");
+            throw new RpcException(new Status(StatusCode.Internal, e.Message));
+        }
+    }
+
+    public override async Task<GetNodesResponse> GetNodes(GetNodesRequest request, ServerCallContext context)
+    {
+        try
+        {
+            var nodes = new List<Data.Models.Node>();
+            if (request.IncludeUnmanaged)
+            {
+                nodes = await _nodeRepository.GetAll();
+            }
+            else
+            {
+                nodes = await _nodeRepository.GetAllManagedByNodeGuard();
+            }
+
+            var mappedNodes = nodes.Select(x => _mapper.Map<Nodeguard.Node>(x)).ToList();
+
+            var response = new GetNodesResponse()
+            {
+                Nodes = {mappedNodes}
+            };
+            return response;
+        }
+        catch (Exception e)
+        {
+            _logger?.LogError(e, "Error getting nodes through gRPC");
+
+            throw new RpcException(new Status(StatusCode.Internal, e.Message));
+        }
+    }
+
+    public override async Task<OpenChannelResponse> OpenChannel(OpenChannelRequest request, ServerCallContext context)
```

Comment:
I think we might need to refactor in the future, This code is repeated in some other parts of the app

Reply:
Invoking the job? Yeah, it can be refactored